morpheus/service/vdb/milvus_vector_db_service.py

```diff
@@ -217,6 +384,158 @@ async def similarity_search(self,
 
         return outputs
 
+    def update(self, data: list[typing.Any], **kwargs: dict[str, typing.Any]) -> dict[str, typing.Any]:
+        """
+        Update data in the collection.
+
+        Parameters
+        ----------
+        data : list[typing.Any]
+            Data to be updated in the collection.
+        **kwargs : dict[str, typing.Any]
+            Extra keyword arguments specific to upsert operation.
+
+        Returns
+        -------
+        dict[str, typing.Any]
+            Returns result of the updated operation stats.
+        """
+
+        if not isinstance(data, list):
+            raise RuntimeError("Data is not of type list.")
+
+        result = self._client.upsert(collection_name=self._name, entities=data, **kwargs)
+
+        self._collection.flush()
+
+        return self._update_delete_result_to_dict(result=result)
+
+    def delete_by_keys(self, keys: int | str | list, **kwargs: dict[str, typing.Any]) -> typing.Any:
+        """
+        Delete vectors by keys from the collection.
+
+        Parameters
+        ----------
+        keys : int | str | list
+            Primary keys to delete vectors.
+        **kwargs :  dict[str, typing.Any]
+            Extra keyword arguments specific to the vector database implementation.
+
+        Returns
+        -------
+        typing.Any
+            Returns result of the given keys that are deleted from the collection.
+        """
+
+        result = self._client.delete(collection_name=self._name, pks=keys, **kwargs)
+
+        self._collection.load()
+
+        return result
+
+    def delete(self, expr: str, **kwargs: dict[str, typing.Any]) -> dict[str, typing.Any]:
+        """
+        Delete vectors from the collection using expressions.
+
+        Parameters
+        ----------
+        expr : str
```

Comment:
Is this just like a regex? if its something custom, can you put a link in the doc string.

Reply:
It's not exactly a regex, but it is similar as shown below.
`expr = "book_id in [0,1]"`

here is the doc: https://milvus.io/docs/delete_data.md